apps/passport-server/src/routing/server.ts

```diff
@@ -62,15 +64,35 @@ export async function startServer(
         }
       );
 
-      const server = app.listen(port, () => {
-        logger(`[INIT] HTTP server listening on port ${port}`);
-        sendEvent(context, EventName.SERVER_START);
-        resolve({ server, app });
-      });
+      if (process.env.IS_LOCAL_HTTPS) {
+        const httpsOptions = {
+          key: fs.readFileSync("../certificates/dev.local-key.pem"),
+          cert: fs.readFileSync("../certificates/dev.local.pem")
+        };
 
-      server.on("error", (e: Error) => {
-        reject(e);
-      });
+        const server = https
+          .createServer(httpsOptions, app)
+          .listen(port, () => {
+            logger(
+              `[INIT] Local HTTPS server listening on https://dev.local:3002`
```

Comment:
how does `dev.local` resolve to properly? did you need to configure anything for that to work? also make sure to not hardcode the `port` in strings here, use the `port` variable

Reply:
Here is the setup for [dev.local](https://github.com/emergenceland/zupass/blob/test/bot.md#6-optional-hot-reloading-for-tg-development). Also resolved by 693e1c919770c8603b25478a5025748d1e522ec3